### DISCUSSION TITLE
Fetching of accounts in foreground

### DISCUSSION THREAD

**Discussion by smoneyan** [Noovolari/leapp#462] [General]
Hi, 

I think fix for [this issue](https://github.com/Noovolari/leapp/issues/459) made the fetching of accounts in foreground after [v0.18.5](https://github.com/Noovolari/leapp/releases/tag/v0.18.5)

Any specific reason why the fetching of accounts was made foreground ? 

What has happened is, everyday after i login, i've to wait for atleast 2-5 mins to fetch all the accounts before i start using leapp. The whole screen is disabled until the fetching of all the accounts is complete. 

<img width="304" alt="image" src="https://github.com/Noovolari/leapp/assets/837733/324c18b2-c6a4-45b6-91f8-4056100139e0">

Is there any work around to make it in background ?

**Comment by pethron** [comment]
Hello @smoneyan. You're correct; we introduced this process to fix the throttling error when retrieving the accounts.
Since the bug was causing significant issues, we compromised on the complexity and made it a foreground process; otherwise, we had to update the logic where the retrieved session could be seen, and the fix would have to wait too long.

There's no workaround to make it background in the current version (only use v0.18.4, but you'll go back to the throttling error). However, we are planning to make it a background process.

We can't commit to a date and need to estimate how complex this can become, but we can use this discussion (or open an issue) to gather interest and see how much people are affected to help decide the priority.
